File: ABC/abc086/c/main.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int N; cin >> N;
    int t[N + 1], x[N + 1], y[N + 1];
    t[0] = x[0] = y[0] = 0;
    for (int i = 1; i <= N; ++i) {
        cin >> t[i] >> x[i] >> y[i];
    }

    for (int i = 0; i < N; ++i) {
        int dt = t[i + 1] - t[i];
        int dist = abs(x[i + 1] - x[i]) + abs(y[i + 1] - y[i]);
        if (dt < dist || dt % 2 != dist %2) {
            cout << "No" << endl;
            return 0;
        }
    }

    cout << "Yes" << endl;
    return 0;
}

File: ABC/abc086/c/main.py
#!/usr/bin/env python

n = int(input())
pos = [list(map(int, input().split())) for i in range(n)]

prev_t, prev_x, prev_y = 0, 0, 0

for i in range(n):
    t, x, y = pos[i][0], pos[i][1], pos[i][2]

    lag = t - prev_t
    distance = abs(x - prev_x) + abs(y - prev_y)
    if distance > lag or lag % 2 != distance % 2:
        print("No")
        exit()

    prev_t, prev_x, prev_y = t, x, y

print("Yes")
